packages/nodes-base/nodes/Totp/Totp.node.ts

```diff
@@ -0,0 +1,169 @@
+import type {
+	IExecuteFunctions,
+	INodeExecutionData,
+	INodeType,
+	INodeTypeDescription,
+} from 'n8n-workflow';
+
+import { NodeOperationError } from 'n8n-workflow';
+
+import OTPAuth from 'otpauth';
+
+export class Totp implements INodeType {
+	description: INodeTypeDescription = {
+		displayName: 'TOTP',
+		name: 'totp',
+		icon: 'fa:fingerprint',
+		group: ['transform'],
+		version: 1,
+		subtitle: '={{ $parameter["operation"] }}',
+		description: 'Generate a time-based one-time password',
+		defaults: {
+			name: 'TOTP',
+		},
+		inputs: ['main'],
+		outputs: ['main'],
+		credentials: [
+			{
+				name: 'totpApi',
+				required: true,
+			},
+		],
+		properties: [
+			{
+				displayName: 'Operation',
+				name: 'operation',
+				type: 'options',
+				noDataExpression: true,
+				options: [
+					{
+						name: 'Generate Secret',
+						value: 'generateSecret',
+						action: 'Generate secret',
+					},
+				],
+				default: 'generateSecret',
+			},
+			{
+				displayName: 'Additional Options',
```

Comment:
Should it be just an 'Options'?

Reply:
Not sure what the convention is? Changed it just in case.